pkg/repos/get.go

```diff
@@ -36,20 +43,81 @@ func (n noopRuntime) Setup(_ context.Context, _, _ string, _ []string) ([]string
 }
 
 type Manager struct {
-	storageDir string
-	gitDir     string
-	runtimeDir string
-	runtimes   []Runtime
+	storageDir     string
+	gitDir         string
+	runtimeDir     string
+	credHelperDirs credentials.CredentialHelperDirs
+	runtimes       []Runtime
 }
 
 func New(cacheDir string, runtimes ...Runtime) *Manager {
 	root := filepath.Join(cacheDir, "repos")
 	return &Manager{
-		storageDir: root,
-		gitDir:     filepath.Join(root, "git"),
-		runtimeDir: filepath.Join(root, "runtimes"),
-		runtimes:   runtimes,
+		storageDir:     root,
+		gitDir:         filepath.Join(root, "git"),
+		runtimeDir:     filepath.Join(root, "runtimes"),
+		credHelperDirs: credentials.GetCredentialHelperDirs(cacheDir),
+		runtimes:       runtimes,
+	}
+}
+
+func (m *Manager) SetUpCredentialHelpers(ctx context.Context, cliCfg *config.CLIConfig, env []string) error {
```

Comment:
Curious why this is necessary and we can't use the built-in Go runtime support.

Reply:
The credential helpers aren't packaged as a tool so I figured it made more sense to handle the building of them separately (especially since it's one repo, but with several binaries that can be built from it). This does ultimately use our Go runtime in order to build it.

Is there a refactoring you think I could do to simplify and remove some of this credential helper-specific build stuff? I'm open to changing it if you think it should be different.

Reply:
Can it be packaged as a tool? Then this code would just be running the tool, which would build it.

Reply:
Well we don't want to actually run it here, only build it. But I'll see what I can do

Reply:
Having tried some things and thought more about this, I do think it is best to keep it as it is in this PR. While it may decrease the amount of code if I were to package these as tools and find ways to build and then find the built binaries when the credential store code needs to use them, I think the resulting code would be less clear and harder to maintain. Handling these separately is best, imo.

Reply:
OK. Thanks for looking into it.